src/cascadia/CommandlineArgs/CommandlineArgs.vcxproj

```diff
@@ -0,0 +1,119 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project DefaultTargets="Build" ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup Label="Globals">
+    <ProjectGuid>{2658e9b0-bd84-48b7-bb5e-3d18f4df07ba}</ProjectGuid>
+    <Keyword>Win32Proj</Keyword>
+    <RootNamespace>Microsoft.Terminal.CommandlineArgs</RootNamespace>
+    <ProjectName>CommandlineArgs</ProjectName>
+    <TargetName>CommandlineArgs</TargetName>
+    <WindowsTargetPlatformMinVersion>10.0.17763.0</WindowsTargetPlatformMinVersion>
+    <ConfigurationType>StaticLibrary</ConfigurationType>
+    <SubSystem>Console</SubSystem>
+    <OpenConsoleUniversalApp>true</OpenConsoleUniversalApp>
+  </PropertyGroup>
+  <Import Project="..\..\..\common.openconsole.props" Condition="'$(OpenConsoleDir)'==''" />
+  <Import Project="$(OpenConsoleDir)src\cppwinrt.build.pre.props" />
+  <ItemDefinitionGroup>
+
+    <ClCompile>
+      <!-- For CLI11: It uses dynamic_cast to cast types around, which depends
+      on being compiled with RTTI (/GR). -->
+      <RuntimeTypeInfo>true</RuntimeTypeInfo>
+    </ClCompile>
+  </ItemDefinitionGroup>
+
+  <!-- ========================= Headers ======================== -->
+  <ItemGroup>
+    <ClInclude Include="AppCommandlineArgs.h" />
+    <ClInclude Include="Commandline.h" />
+    <ClInclude Include="pch.h" />
+  </ItemGroup>
+  <!-- ========================= Cpp Files ======================== -->
+  <ItemGroup>
+    <ClCompile Include="AppCommandlineArgs.cpp" />
+    <ClCompile Include="Commandline.cpp" />
+    <ClCompile Include="pch.cpp">
+      <PrecompiledHeader>Create</PrecompiledHeader>
+    </ClCompile>
+  </ItemGroup>
+  <!-- ========================= idl Files ======================== -->
+  <ItemGroup>
+  </ItemGroup>
+  <!-- ========================= Misc Files ======================== -->
+  <ItemGroup>
+    <None Include="packages.config" />
+  </ItemGroup>
+  <!-- ========================= Project References ======================== -->
+  <ItemGroup>
+    <!--
+      the packaging project won't recurse through our dependencies, you have to
+      make sure that if you add a cppwinrt dependency to any of these projects,
+      you also update all the consumers
+    -->
+    <ProjectReference Include="$(OpenConsoleDir)src\types\lib\types.vcxproj" />
+    <ProjectReference Include="$(OpenConsoleDir)src\cascadia\WinRTUtils\WinRTUtils.vcxproj">
+      <Project>{CA5CAD1A-039A-4929-BA2A-8BEB2E4106FE}</Project>
+      <ReferenceOutputAssembly>false</ReferenceOutputAssembly>
+    </ProjectReference>

```

Comment:
Wait... you might be able to get rid of WinRTUtils no? I don't think you're using the macros.

Reply:
wait shoot I am. AppCommandlineArgs uses `RS_A` to get the descriptions for args. That'll inevitably explode when I try doing this. 

@DHowett do you have any clever ideas for a way around this?

Reply:
G A H 

Yep this will crash immediately. I'll work on some more clever solutions in the morning.

Reply:
Okay, the Terrible solution I have is to have a `.../CommandlineArgs/Resources/en-us/Resources.resw`, and include it manually in each of TSE and `TerminalApp`. That'll get the resources added to the correct libraries, so `RS_A` will still work, regardless of which dll it's being called from. 

@DHowett Will that still work for localization?

Reply:
I think that it will require both key instances to be localized separately? I'm not 100% sure actually. It's by _file path_ actually . . . so maybe it actually DOES work ...